test/test_args.py

```diff
@@ -0,0 +1,165 @@
+import sys
+import os
+import unittest
+import tempfile
+
+from muspinsim.__main__ import main
+
+
+def set_sys_argv(vals):
+    sys.argv[1:] = vals
+
+
+def run_experiment(inp_string, inp_file_path, cmd_args):
+    with open(inp_file_path, "w") as f:
+        f.write(inp_string)
+    set_sys_argv([inp_file_path] + cmd_args)
+    main()
+
+
+class TestCommandLineArgs(unittest.TestCase):
+    def test_main_default_args(self):
+        with tempfile.TemporaryDirectory() as tmp_dir:
+            cmd_args = []
+            inp_file_path = os.path.join(tmp_dir, "custom_args.in")
+            run_experiment(
+                """
+name
+    test_1
+spins
+    mu H
+zeeman 1
+    1 0 0
+""",
+                inp_file_path,
+                cmd_args,
+            )
+
+            # self.assertTrue(os.path.exists("{0}/custom_args.log".format(tmp_dir)))
+            self.assertTrue(os.path.exists("{0}/test_1.dat".format(tmp_dir)))
+
+    def test_main_custom_args(self):
+        with tempfile.TemporaryDirectory() as tmp_dir:
+            with tempfile.TemporaryDirectory() as out_tmp_dir:
+                cmd_args = [
+                    "-o",
+                    "{0}".format(out_tmp_dir),
+                    "-l",
+                    "{0}/new_test.log".format(out_tmp_dir),
+                ]
+                inp_file_path = os.path.join(tmp_dir, "custom_args.in")
+                run_experiment(
+                    """
+name
+    test_2
+spins
+    mu H
+zeeman 1
+    1 0 0
+""",
+                    inp_file_path,
+                    cmd_args,
+                )
+
+                # self.assertTrue(os.path.exists("{0}/new_test.log".format(out_tmp_dir)))
```

Comment:
There seems to be a bug where the log file doesn't get created when this test is run.

Reply:
I see commented lines like this in every test - do they all have this problem?